package/MDAnalysis/topology/LAMMPSParser.py

```diff
@@ -188,97 +258,122 @@ def iterdata(self):
         with openany(self.filename) as f:
             for line in f:
                 line = line.partition('#')[0].strip()
-                if line:
+                if line:  # reading non-empty lines
                     yield line
 
     def grab_datafile(self):
         """Split a data file into dict of header and sections
 
         Returns
         -------
-        header - dict of header section: value
-        sections - dict of section name: content
+        header: dict of str
+            A dict in which a key and value pair is a 'header' name and
+            its value in str foramt.
+        sections: dict of list of str
+            A dict in which a key and value pair is a 'section' name and
+            a list of strings (relevenat lines of the data file).
         """
         f = list(self.iterdata())
 
+        # finding the starting line of each section:
         starts = [i for i, line in enumerate(f)
                   if line.split()[0] in SECTIONS]
+        # adding None to properly handle lines in the last SECTION below
         starts += [None]
 
         header = {}
+
+        # lines before the first SECTION found in the file are about HEADERS:
         for line in f[:starts[0]]:
             for token in HEADERS:
                 if line.endswith(token):
                     header[token] = line.split(token)[0]
                     continue
 
-        sects = {f[l]:f[l+1:starts[i+1]]
+        sects = {f[l]: f[l+1:starts[i+1]]  # f[l] is indeed a SECTION name
                  for i, l in enumerate(starts[:-1])}
 
         return header, sects
 
     @staticmethod
     def _interpret_atom_style(atom_style):
-        """Transform a string description of atom style into a dict
-
-        Required fields: id, type, x, y, z
-        Optional fields: resid, charge
-
-        eg: "id resid type charge x y z"
-        {'id': 0,
-         'resid': 1,
-         'type': 2,
-         'charge': 3,
-         'x': 4,
-         'y': 5,
-         'z': 6,
-        }
+        """Map field names in a given `atom_style` to their corresponding
```

Comment:
Also is it possible to have custom atom styles in LAMMPs? if so how are we handling them as have removed flexible atom style parsing.

Reply:
@hmacdope Yes, the format of the dump files is _extremely_ customizable (the order of columns listed can change, plus the user can define their own variables and dump those).

Reply:
@hmacdope Yes, the format of the dump files is _extremely_ customizable (the order of columns listed can change, plus the user can define their own variables and dump those).

Reply:
Please see my comment above. dump styles (trajectory files) are customizable but data styles (topology files) are pre-defined.

Reply:
Sorry for further question but does this also include for the `hybrid` style?